app/views/main/dashboard.html.erb

```diff
@@ -15,9 +15,38 @@
           <td>
             <div class="btn-group" role="group" aria-label="Validate or Remove Teacher">
               <%= button_to("Validate", validate_path(:id => teacher.id),
-                 class: 'btn btn-success', type: 'button') %>
-              <%= button_to("Deny", deny_path(:id => teacher.id),
-                 class: 'btn btn-danger', type: 'button') %>
+                class: 'btn btn-success', type: 'button') %>
+              <span>
+                <button class="btn btn-danger" type="button" data-toggle="modal" data-target="<%= "#denialModal" + teacher.id.to_s %>" >
+                  Deny
+                </button>
+              </span>
+              <div class="modal fade" id="<%= "denialModal" + teacher.id.to_s %>" tabindex="-1" role="dialog" aria-labelledby="denialModalLabel" aria-hidden="true">
```

Comment:
This is really nice!  Nicer than I was expecting. :) And good use of Bootstrap!

Two things:
* It would be great if we could move this modal code into a partial.
* I think we can refactor the code to only need one instance of the HTML, which should probably be included outside of the table code. See [the bootstrap example](https://getbootstrap.com/docs/4.3/components/modal/#varying-modal-content) for "varying" content. (Think about what is unique to each version and where to put that info.)